### PR TITLE
Fix Form styles.

### PR DESCRIPTION
**Pull request by jnorris-carecloud**
**Changes**:
- Fix AlfrescoForm spacing (margin, padding).
- Collections:
  - Make action buttons (up, down, remove) smaller.
  - Use `inverted` variant for "add" button and fix styling.
  - Avoid an extra container.
- Use `required` and `disabled` properties in label props.
- Fix warning about `formData` in `StringField`.

### CODE REVIEW

**Review by nbroda-carecloud — APPROVED** [review]

File: src/components/alfresco/AlfrescoForm.jsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ class AlfrescoForm extends Component {
         const component = this.props.properties.filter(property => property.name === containerKey)[0];
 
         return (
-          <Grid key={containerIndex} item xs={12}>
+          <Grid item key={containerIndex} xs={12} style={{ marginBottom: 24 }}>
```
Comment (amartin-carecloud) [inline src/components/alfresco/AlfrescoForm.jsx]: can we move the inline styles to classes?

File: src/components/fields/SchemaField.js [hunk 2]
```diff
@@ -185,6 +187,8 @@ function SchemaFieldRender(props) {
     case 'array':
       displayLabel =
         isMultiSelect(schema, definitions) || isFilesArray(schema, uiSchema, definitions);
+      // If it's an array of objects, use a div element
+      useDiv = !!schema.items && schema.items.type === 'object';
```
Comment (amartin-carecloud) [inline src/components/fields/SchemaField.js]: Is it more readable to wrap that expression in `( ... )` ?

File: src/components/alfresco/AlfrescoForm.jsx (hunk 1 above)
Comment (jnorris-carecloud) [reply src/components/alfresco/AlfrescoForm.jsx]: Yes, good point.

File: src/components/fields/SchemaField.js (hunk 2 above)
Comment (jnorris-carecloud) [reply src/components/fields/SchemaField.js]: Yes, probably if I do:

```js
useDiv = !!schema.items && (schema.items.type === 'object');
```

File: src/components/fields/SchemaField.js [hunk 3]
```diff
@@ -186,6 +188,9 @@ function SchemaFieldRender(props) {
       displayLabel =
         isMultiSelect(schema, definitions) || isFilesArray(schema, uiSchema, definitions);
```
Comment (jnorris-carecloud) [inline src/components/fields/SchemaField.js:189]: `prettier` is removing the parenthesis around these expressions.

File: src/components/fields/ArrayField.js [hunk 4]
```diff
@@ -43,15 +45,18 @@ function DefaultArrayItem(props) {
...
   };
 
   return (
     <div key={props.index} className={props.className}>
       <div className={props.hasToolbar ? 'col-xs-9' : 'col-xs-12'}>{props.children}</div>
 
       {props.hasToolbar && (
-        <div className="col-xs-3 array-item-toolbox">
-          <div className="btn-group" style={{ display: 'flex', justifyContent: 'flex-end' }}>
+        <div className="col-xs-3 array-item-toolbox" style={{ marginBottom: '10px' }}>
```
Comment (amartin-carecloud) [inline src/components/fields/ArrayField.js:58]: Inline styles necessary?

File: src/components/fields/ArrayField.js [hunk 5]
```diff
@@ -153,7 +162,11 @@ function DefaultNormalArrayFieldTemplate(props) {
       </div>
 
       {props.canAdd && (
-        <AddButton onClick={props.onAddClick} disabled={props.disabled || props.readonly} />
+        <AddButton
+          // numberOfItems={props.items ? props.items.length : 0}
```
Comment (amartin-carecloud) [inline src/components/fields/ArrayField.js:166]: remove commented code

File: src/components/fields/ArrayField.js [hunk 6]
```diff
@@ -122,7 +127,11 @@ function DefaultFixedArrayFieldTemplate(props) {
...
```
Comment (amartin-carecloud) [inline src/components/fields/ArrayField.js:131]: remove commented code

**Review by hcuneo-carecloud — APPROVED** [review]

**Pull request merged**